openURL/src/functions.php

```diff
@@ -132,197 +56,20 @@ function token_fetch($clientID, $secret) {
-	
-	/* uncomment for debugging
-	echo "    </br>";
-	echo "    Updated ETag: " . $etag . "</br>";
-	echo "</br>";
-	*/
-
-	return $etag;
-}
-      
-function make_resource($shortCode, $title, $resource_type, $isbn, $token, $lcn, $full_name, $edition, $publisher_name, $web_addresses) {
-
-	$uuid = guidv4();
-	$url = 'https://rl.talis.com/3/' . $shortCode . '/resources';
-	 
-	if (!empty ($full_name)) {
-		$full_name  = '"' . $full_name . '"';
-	} else {
-		$full_name = "null";
-	}	 
-	 
-	if (!empty ($isbn)) {
-		$isbn = '["' . $isbn . '"]';
-	} else {
-		$isbn = "null";
-	}
-			
-	if (!empty ($edition)) {
-		$edition = '"' . $edition . '"';
-	} else {
-		$edition = "null";
-	}
-		 
-	if (!empty ($title)) {
-		$title = '"' . $title . '"';
-	} else {
-		$title = "null";
-	}
-	
-	if (!empty ($lcn)) {
-		$lcn = '"' . $lcn . '"';
-	} else {
-		$lcn = "null";
-	}
-		
-	if (!empty ($publisher_name)) {
-		$publisher_name = '["' . $publisher_name . '"]';
-	} else {
-		$publisher_name = "null";
-	}
-	
-	if (!empty ($resource_type)) {
-		$resource_type = '"' . $resource_type . '"';
-	} else {
-		$resource_type = "null";
-	}
-			 
-	if (!empty ($web_addresses)) {
-		$web_addresses = '["' . $web_addresses . '"]';
-	} else {
-		$web_addresses = "null";
-	}
-					 
-	$body = '{
-		"data": {
-		  "id": "' . $uuid . '",
-		  "type": "resources",
-		  "attributes": {
-            "authors": [
-                {
-                    "full_name": ' . $full_name . '          
-                }
-						],
-            "isbn13s": ' . $isbn . ',
-            "edition": ' . $edition . ',
-            "lcn": ' . $lcn . ',
-            "publisher_name": ' . $publisher_name . ',
-            "resource_type": ' . $resource_type . ',
-            "title": ' . $title . ',
-            "web_addresses": ' . $web_addresses . '
-						},
-		  "links": {},
-		  "meta": {},
-		  "relationships": {}
-				}
-	  }';
-	
-	//var_export($body);
-	
-	$ch = curl_init();
-	
-	curl_setopt($ch, CURLOPT_URL, $url);
-	curl_setopt($ch, CURLOPT_CUSTOMREQUEST, 'POST');
-	curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
-	curl_setopt($ch, CURLOPT_HTTPHEADER, array(
-		
-		"Authorization: Bearer $token",
-		'Cache-Control: no-cache'
-	
-	));
-	curl_setopt($ch, CURLOPT_POSTFIELDS, $body);
-
-	$output = curl_exec($ch);
-	$info = curl_getinfo($ch, CURLINFO_HTTP_CODE);
-	//echo $info;
-	$output_json = json_decode($output);
-	curl_close($ch);
-	if ($info !== 200){
-		echo "<p>ERROR: There was an error creating resource for $isbn:</p><pre>" . var_export($output, true) . "</pre>";
-	}
-	
-	return $uuid;
-}
-
 function update_resource($shortCode, $token, $resourceID) {
-
 	$url = 'https://rl.talis.com/3/' . $shortCode . '/resources/' . $resourceID;
 	 
-	$body = '{
-				"data": {
-					"type": "resources",
-					"id": "' . $resourceID . '",
-					"attributes": {
-						"online_resource": {
-							"source": "open_url"
-						}
-					}
-				}
-			}';
-	
+	$body = json_encode([
```

Comment:
Using json_encode means we don't need to worry about formatting the json ourselves, PHP will do that for us.  https://www.php.net/manual/en/function.json-encode.php

This leaves us with an array that's easy to read, and which will syntax highlight nicely in an IDE.